src/main.js

```diff
@@ -74,14 +75,45 @@ let load = loader({
 
   github: {
     requires: ['cfg'],
-    setup: ({cfg}) => {
+    setup: async ({cfg}) => {
       let github = new Github({
         promise: Promise,
       });
-      if (cfg.github.credentials.token) {
-        github.authenticate(cfg.github.credentials);
-      }
-      return github;
+
+      // This object insures that the authentication is delayed until we need it.
+      // Also, the authentication happens not just once in the beginning, but for each request.
+      return {
```

Comment:
This is pretty long to include in `main.js`.  Please move this code to a new module (`src/github_auth.js` or something like that).